src/VtkGeometryRenderer.js

```diff
@@ -0,0 +1,112 @@
+/* global window */
+
+import React                from 'react';
+
+// import sizeHelper           from 'paraviewweb/src/Common/Misc/SizeHelper';
+
+import vtkOpenGLRenderWindow           from 'vtk.js/Sources/Rendering/OpenGL/RenderWindow';
+import vtkSynchronizableRenderWindow   from 'vtk.js/Sources/Rendering/Misc/SynchronizableRenderWindow';
+import vtkRenderWindowInteractor       from 'vtk.js/Sources/Rendering/Core/RenderWindowInteractor';
+
+const SYNCHRONIZATION_CONTEXT_NAME = 'pvwLocalRenderingContext';
+
+export default React.createClass({
+
+  displayName: 'VtkGeometryRenderer',
+
+  propTypes: {
+    className: React.PropTypes.string,
+    client: React.PropTypes.object,
+    viewId: React.PropTypes.number,
+    interactionTimout: React.PropTypes.number,
+    connection: React.PropTypes.object,
+    showFPS: React.PropTypes.bool,
+    style: React.PropTypes.object,
+  },
+
+  getDefaultProps() {
+    return {
+      className: '',
+      showFPS: false,
+      style: {},
+      viewId: -1,
+      interactionTimout: 500,
+    };
+  },
+
+  componentWillMount() {
+    // Make sure we monitor window size if it is not already the case
+    console.log('Mounting VtkGeometryRenderer');
+    // sizeHelper.startListening();
+  },
+
+  componentDidMount() {
+    console.log('VtkGeometryRenderer is now mounted');
+
+    // Get our hands on the default synchronization context and tell how to fetch data arrays
+    const synchCtx = vtkSynchronizableRenderWindow.getSynchronizerContext(SYNCHRONIZATION_CONTEXT_NAME);
+    synchCtx.setFetchArrayFunction(this.props.client.VtkGeometryDelivery.getArray);
+
+    const container = this.rootContainer;
+
+    // container.className = style.renderWindowContainer;
+    container.setAttribute('data-view-id', this.props.viewId);
+
+    // VTK renderWindow/renderer
+    const renderWindow = vtkSynchronizableRenderWindow.newInstance({ synchronizerContextName: SYNCHRONIZATION_CONTEXT_NAME });
+
+    // OpenGlRenderWindow
+    const openGlRenderWindow = vtkOpenGLRenderWindow.newInstance();
+    openGlRenderWindow.setContainer(container);
+    renderWindow.addView(openGlRenderWindow);
+
+    // Interactor
+    const interactor = vtkRenderWindowInteractor.newInstance();
+    interactor.setView(openGlRenderWindow);
+    interactor.initialize();
+    interactor.bindEvents(container);
+
+    this.props.client.VtkGeometryDelivery.onViewChange((data) => {
+      console.log('Received scene desciption:');
+      console.log(data);
+      renderWindow.synchronize(data[0]);
+    });
+
+    this.props.client.VtkGeometryDelivery.addViewObserver(this.props.viewId).then((successResult) => {
+      console.log(`Subscribe to view ${this.props.viewId} succeeded`);
+      console.log(successResult);
```

Comment:
Since you are putting that information as attribute `container.setAttribute('data-view-id', this.props.viewId);` this should be managed via state and render function.